internal/jobs/noop_job.go

```diff
@@ -14,28 +12,26 @@ type NoopJobArgs struct {
 	ReservationID int64 `json:"reservation_id"`
 }
 
+// Unmarshall arguments and handle error
 func HandleNoop(ctx context.Context, job dejq.Job) error {
-	ctxLogger := ctxval.Logger(ctx)
-	ctxLogger.Debug().Msg("Started no operation job")
-
 	args := NoopJobArgs{}
-	err := job.Decode(&args)
-	if err != nil {
-		ctxLogger.Error().Err(err).Msg("unable to decode arguments")
-		return fmt.Errorf("unable to decode args: %w", err)
-	}
-	logger := ctxLogger.With().Int64("reservation", args.ReservationID).Logger()
-	logger.Info().Interface("args", args).Msg("Processing no operation job")
-
-	// do nothing and update status
-	rDao, err := dao.GetReservationDao(ctx)
-	if err != nil {
-		return fmt.Errorf("cannot get reservation DAO: %w", err)
-	}
-	err = rDao.Finish(ctx, args.ReservationID, true, "Finished")
-	if err != nil {
-		return fmt.Errorf("cannot update reservation status: %w", err)
-	}
+	decodeJob(ctx, job, args)
+	ctx = contextLogger(ctx, job.Type(), args, args.AccountID, args.ReservationID)
+
+	jobErr := handleNoop(ctx, &args)
+
+	finishJob(ctx, args.ReservationID, jobErr)
+	return jobErr
+}
+
+// Job logic, when error is returned the job status is updated accordingly
+func handleNoop(ctx context.Context, args *NoopJobArgs) error {
+	// status updates before and after the code logic
+	updateStatusBefore(ctx, args.ReservationID, "No operation started")
+	defer updateStatusAfter(ctx, args.ReservationID, "No operation finished", 1)
+
+	// do nothing and return no error
+	time.Sleep(5 * time.Second)
 
 	return nil
 }
```

Comment:
All of these changes are improvements to error handling, I feel like a separate method with an explicit error is a good way to ensure error handling is always correct. Also using `defer` allows me to ensure that we will never forget to set status and update step even when there is an error. All the other jobs have the very same changes, just review this one and then read the others to ensure I did not do another copy&paste mistake :-)